src/solve.py

```diff
@@ -103,8 +111,29 @@ def main():
 
     solution = model.solve(TIMEOUT)
 
-    loader.write_solution(
-        solution,
+    combined_constraints = []
+    for constraint in constraints:
+        combined_constraints.append(constraint.KEY)
+    for objective in objectives:
+        combined_constraints.append(objective.KEY)
+
+    constraint_index = get_combined_indices_string(
+        original_constraints, combined_constraints
+    )
+    solution_name = create_solutionNameData(startYear, startMonth, constraint_index)
+
+    loader.write_solution(solution, solution_name)
+
+
+def create_solutionNameData(startYear, startMonth, constraintIndex):
+    return f"{startYear}_{startMonth}_CON{constraintIndex}"
+
+
+def get_combined_indices_string(original_constraints, combined_constraints):
+    return "".join(
+        str(original_constraints.index(item))
+        for item in combined_constraints
+        if item in original_constraints
     )
 
```

Comment:
I think you only need the case ID, now called the planning unit, and the start and end dates. The constraints shouldn't be part of the file name.